### DISCUSSION TITLE
Paste files from clipboard

### DISCUSSION THREAD

**Discussion by plundration** [sxyazi/yazi#1797] [4. Ideas]
Is it currently possible to somehow paste an image from the clipboard? Like when copying from a browser? I know there is a yank-to-clipboard plugin, but I couldn't find anything about pasting. This would be a great feature to add, I use it a lot in graphical file explorers.

**Comment by AnirudhG07** [comment]
It would be better to extend from not just image but to any file(including images) present in clipboard to here.  ill look into it, sounds nice(afaik it is not implemented as a plugin)

**Reply by Vaisakhkm2625** [reply]
i assume supporting a lot of filetypes also going to be difficult, especially on linux, 
- in wayland, i remember issue of  doesn't allow pasting to to unfocused windows.. wl-clipboard does some witchcraft workarounds under the hood though..
- file formats/mimetypes for same type of file differ between different types for file managers....
i copied exact same file from `thunar` and `Dolphin`,
```sh
❮ wl-paste --list-types
text/uri-list
x-special/gnome-copied-files
text/plain;charset=utf-8
UTF8_STRING
```

```sh
❯ wl-paste --list-types
text/uri-list
application/x-kde4-urilist
application/vnd.portal.filetransfer
```

 as you can see, gtk file managers add `x-special/gnome-copied-files` while dolphin adds `application/x-kde4-urilist`

**Comment by Tahinli** [comment]
Do we have any update about it ?

**Comment by chea-vuthearith** [comment]
if anyone still needs this
```
# keymap.toml

[[manager.prepend_keymap]]
on   = "P"
run  = "shell -- wl-paste > pasted-image.png"
desc = "Paste image from clipboard"
```

**Reply by sohanemon** [reply]
![image](https://github.com/user-attachments/assets/249b0b43-faf4-4b90-b275-8bc15992d1e2)
its not working in tmux for me

**Reply by chea-vuthearith** [reply]
apologies for not mentioning but the cmd above is for wayland, if youre on x11 maybe try replacing wl-paste with

xclip -selection clipboard -o

**Reply by sohanemon** [reply]
I'm on Hyprland; it's based on Wayland.

I found it works if I'm outside of tmux, but it fails to work inside tmux.

**Comment by boydaihungst** [comment]
Here you go, doesn't need to install any 3rd party app: https://github.com/boydaihungst/save-clipboard-to-file.yazi

**Comment by FMVPJet** [comment]
I have developed a plugin, you can give it a try.

https://github.com/FMVPJet/handoff.yazi
